tools/docker/Dockerfile

```diff
@@ -105,8 +105,10 @@ RUN wget -nv https://www.nordicsemi.com/-/media/Software-and-other-downloads/Des
 RUN python3 -m pip -q install --upgrade pip && \
     python3 -m pip -q install \
       setuptools \
-      sphinx_rtd_theme \
-      sphinx \
+      'sphinx_rtd_theme==1.0.0' \
+      'myst-parser==0.18.0' \
+      'linkify-it-py==2.0.0' \
+      'sphinx==5.0.2' \
```

Comment:
Please leave a comment about why the latest version of everything doesn't work.

Reply:
This is the latest version of everything at the time of this PR - the goal is to keep the versions pinned and consistent between RTD (`tools/readthedocs/requirements.txt`) and the Docker image to prevent documentation builds from breaking in the future. 
We can bump as required, keeping `Dockerfile` and the RTD requirements file consistent.

Reply:
It would be nice to have a single master for the version information, does something like the following work?
```
COPY ../../tools/readthedocs/requirements.txt /tmp
RUN python3 -m pip -q install --upgrade pip && \
    python3 -m pip -q install \
      setuptools \
      matplotlib \
      'protobuf<=4' && \
    python3 -m pip -q install \
      nrfutil && \
    python3 -m pip -q install -r /tmp/requirements.txt && \
    rm -rf /root/.cache /tmp/requirements.txt
```

Reply:
Yes, I wanted to do something like that too but the COPY trick never crossed my mind!
I'll update the PR in a bit with this.

Reply:
The trick breaks stuff in our CI with calculating when docker images should be rebuilt, but it should work for your update, and the rest can be fixed after the release.

Please add a comment to requirements.txt that it is used by the Dockerfile, these kind of things are impossible to find for someone who does a drive-by and isn't looking in the right place by accident.

Reply:
This won't work it seems, Docker complains with `COPY failed: forbidden path outside the build context: ../../tools/readthedocs/requirements.txt ()`.

Reply:
Put `tools/readthedocs/requirements.txt` in `tools/docker/files` instead, and change the Dockerfile to:
```
# Requirements referenced from tools/readthedocs.
COPY files/requirements.txt /tmp
# Matplotlib is required for result visualization.
# Nrfutil 6.1.3 does not work with protobuf 4, so install latest 3.x
# Keep the image size down by removing the pip cache when done.
RUN python3 -m pip -q install --upgrade pip && \
    python3 -m pip -q install \
      setuptools \
      matplotlib \
      'protobuf<=4' && \
    python3 -m pip -q install \
      nrfutil && \
    python3 -m pip -q install -r /tmp/requirements.txt && \
    rm -rf /root/.cache /tmp/requirements.txt
```
And put:
```
-r ../docker/files/requirements.txt
```
in `tools/readthedocs/requirements.txt`.